wallet-api/tutorial/Tutorial.md

```diff
@@ -1,35 +1,42 @@
+
+The tutorial has two parts. In part 1 we write the contract, including all the data types we need, validator scripts, and contract endpoints that handle the interactions between wallet and blockchain. In part 2 we show how to test the contract.
+
+# 1. Contract Definition
+
+We need some language extensions and imports:
 
 ```haskell
-{-# LANGUAGE DataKinds         #-}
-{-# LANGUAGE RecordWildCards   #-}
-{-# LANGUAGE TemplateHaskell   #-}
-{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE DeriveGeneric       #-}
 {-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE StandaloneDeriving #-}
-{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StandaloneDeriving  #-}
+{-# LANGUAGE OverloadedStrings   #-}
 {-# OPTIONS_GHC -O0 #-}
 module Tutorial where
-```
 
-We need some imports:
-
-```haskell
-import qualified Language.PlutusTx            as PlutusTx
-import           Ledger
-import           Ledger.Validation            as Validation
-import           Wallet
-import           Wallet.Emulator
+import qualified Language.PlutusTx            as P
+import           Ledger                       (Address, DataScript(..), PubKey(..), RedeemerScript(..), Signature(..), Slot(..), TxId, ValidatorScript(..), Value(..))
+import qualified Ledger                       as L
+import           Ledger.Validation            (PendingTx(..), PendingTxIn(..), PendingTxOut)
+import qualified Ledger.Validation            as V
+import           Wallet                       (EventHandler(..), EventTrigger, Range(Interval, GEQ))
+import qualified Wallet                       as W
+import           Wallet.Emulator              (MockWallet)
 import           Prelude                      hiding ((&&))
 import           GHC.Generics                 (Generic)
 ```
 
-The module imported as `Validation` contains types and functions that can be used in on-chain code. `PlutusTx` lets us translate code between Haskell and Plutus Core (see the [PlutusTx tutorial](https://github.com/input-output-hk/plutus/blob/master/plutus-tx/tutorial/Tutorial.md)). `Wallet.Emulator` covers interactions with the wallet, for example generating the transactions that actually get the crowdfunding contract onto the blockchain.
+The module `Ledger.Validation`, imported as `V`, contains types and functions that can be used in on-chain code. `Language.PlutusTx` lets us translate code between Haskell and Plutus Core (see the [PlutusTx tutorial](https://github.com/input-output-hk/plutus/blob/master/plutus-tx/tutorial/Tutorial.md)). `Ledger` has data types for the ledger model and `Wallet` is the wallet API. `Wallet.Emulator` covers interactions with the wallet, for example generating the transactions that actually get the crowdfunding contract onto the blockchain.
```

Comment:
I think a relative link for the link to the plutus tx tutorial is better.

Reply:
done